src/http-response-message.js

```diff
@@ -4,31 +4,29 @@ import {Headers} from './headers';
 export class HttpResponseMessage {
   constructor(requestMessage, xhr, responseType, reviver){
     this.requestMessage = requestMessage;
-    this.statusCode = xhr.status;
     this.response = xhr.response || xhr.responseText;
-    this.isSuccess = xhr.status >= 200 && xhr.status < 400;
+    this.statusCode = xhr.status;
     this.statusText = xhr.statusText;
     this.reviver = reviver;
-    this.mimeType = null;
+    
+    this.isSuccess = this.statusCode >= 200 && this.statusCode < 400;
 
-    if(xhr.getAllResponseHeaders){
-      try{
-        this.headers = Headers.parse(xhr.getAllResponseHeaders());
-      }catch(err){
-        //if this fails it means the xhr was a mock object so the `requestHeaders` property should be used
-        if(xhr.requestHeaders) this.headers = { headers:xhr.requestHeaders };
-      }
-    }else {
+    if(xhr.getAllResponseHeaders) {
+      this.headers = Headers.parse(xhr.getAllResponseHeaders());
+    } else {
+      // this is a hack designed to make mocking xhr objects easier
+      // it should be removed as soon as possible
```

Comment:
Was the extra branch that uses `xhr.requestHeaders` just unused by our tests?

If the hack should be removed I'd rather we create an issue for it rather than use a comment that will probably be ignored forever.